File: src/BetterLyrics.KMP/shared/src/commonMain/kotlin/com/jayfunc/betterlyrics/sdk/interfaces/plugins/IPluginContext.kt
package com.jayfunc.betterlyrics.sdk.interfaces.plugins

interface IPluginContext {
    val pluginDirectory: String

    val aiService: IAIService?
    val localizer: ILocalizer

    /**
     * If you are modifying config on the plugin side, please use
     * [BetterLyrics.Sdk.Abstractions.Plugins.PluginBase.config]
     * directly.
     */
    val configurator: IConfigurator
}

File: src/BetterLyrics.DotNet/BetterLyrics.Sdk/Interfaces/Plugins/IPluginContext.cs
namespace BetterLyrics.Sdk.Interfaces.Plugins;

public interface IPluginContext
{
    string PluginDirectory { get; }

    IAIService? AIService { get; }
    ILocalizer Localizer { get; }

    /// <summary>
    ///     If you are modifying config in plugin side, please use
    ///     <see cref="PluginBase{TConfig}.Config" />
    ///     directly.
    /// </summary>
    IConfigurator Configurator { get; }
}